src/libusermode/userhook.cpp

```diff
@@ -187,6 +188,36 @@ static dll_t* get_pending_dll(drakvuf_t drakvuf, drakvuf_trap_info* info, userho
     return nullptr;
 }
 
+int inject_copy_memory(userhook* plugin, drakvuf_t drakvuf,
```

Comment:
This name is incorrect. It doesn't copy memory, but serve as a "page fault" injector. I propose implement a proper `inject_copy_memory` wtith proper arguments (src, dest, buf, etc.) and add another helper `inject_page_load` or smth like that.
Another way would be to rename this function and add proper docstring why this uses `MmCopyVirtualMemory`.

Reply:
No. This injectcs a system call for `MmCopyVirtualMemory`. See my comment below for @icedevml .

With `MmCopyVirtualMemory` injection we really read the contents of source page we would like to load into the small buffer on the stack. This allows to avoid injection of allocate on non-paged pool for this buffer.